scripts/workloads-spin-registry-push.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Check if kubectl is installed
```

Comment:
nit/potential follow-up:  Would be nice to consolidate all of the logic shared by these two scripts.  Looks like everything is the same besides the `kubectl apply -f tests/workloads-{spin-registry-push | docker-build-push}` bit?

Reply:
agree. I didn't want to touch a lot of existing scripts in this PR to avoid breaking workflow other folks are used to when testing changes and to keep the changes contained for easy review.

I can take up another task next to cleanup the scripts a bit and try out dagger functions to see if they help simplify our workflows.

Reply:
+1 to @vdice 's comment

Reply:
would we be fine to add a param to this script to tell which one we want to run now. 

e.g. it would become something like:

./workloads.sh "spin-registry-push-artifacts"

and

./workloads.sh "docker-build-push-artifacts"

Reply:
I've taken a stab at restructuring this a bit as suggested. 

- renamed `workloads.sh` to `deploy-workloads.sh` - this by default deploys `docker build` artifacts, but we can pass a command line argument to deploy `spin registry push` artifacts
- renamed `workloads-delete.sh` to `pod-terminates-test.sh` to reflect the actual purpose of that script
- added `teardown-workloads.sh` to delete all the workloads deployed by `deploy-workloads.sh`

Kindly let me know if this looks any better. happy to hear more feedback and adjust accordingly. kindly let me know (and thank you for spending time to review the code with me here).

Reply:
This looks good to me 👍